File: Qubit Server/dev/src/org/labrad/qubits/channeldata/IqDataBase.java
package org.labrad.qubits.channeldata;

import org.labrad.qubits.channels.IqChannel;

public abstract class IqDataBase implements IqData {

  private IqChannel channel;

  private boolean isDeconvolved = false;

  @Override
  public void setChannel(IqChannel channel) {
    this.channel = channel;
  }

  protected IqChannel getChannel() {
    return channel;
  }

  /**
   * Whether this bit of analog data has been deconvolved.
   */
  @Override
  public boolean isDeconvolved() {
    return isDeconvolved;
  }

  @Override
  public void invalidate() {
    isDeconvolved = false;
  }
  
  protected void setDeconvolved(boolean isDeconvolved) {
    this.isDeconvolved = isDeconvolved;
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/channeldata/IqDataBase.scala
package org.labrad.qubits.channeldata

import org.labrad.qubits.channels.IqChannel

abstract class IqDataBase extends IqData {

  private var channel: IqChannel = _

  @volatile private var _isDeconvolved = false

  override def setChannel(channel: IqChannel): Unit = {
    this.channel = channel
  }

  protected def getChannel(): IqChannel = {
    channel
  }

  /**
   * Whether this bit of analog data has been deconvolved.
   */
  def isDeconvolved(): Boolean = {
    _isDeconvolved
  }

  override def invalidate(): Unit = {
    setDeconvolved(false)
  }

  private[channeldata] def setDeconvolved(isDeconvolved: Boolean): Unit = {
    _isDeconvolved = isDeconvolved
  }
}
